### PR TITLE
fix Issue 17407 - [REG] __traits(compiles) triggers assertion failure

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: test/compilable/interpret3.d [hunk 1]
```diff
@@ -7719,3 +7719,15 @@ int[] f16094b()
 
 enum copy16094a = f16094a();
 enum copy16094b = f16094b();
+
+/**************************************************/
+// https://issues.dlang.org/show_bug.cgi?id=17407
+
+bool foo17404()
```
Comment (UplinkCoder) [inline test/compilable/interpret3.d]: this should probably be foo17407 ?

Comment (UplinkCoder) [inline test/compilable/interpret3.d]: @wa

File: src/ddmd/ctfeexpr.d [hunk 2]
```diff
@@ -1804,7 +1804,7 @@ extern (C++) bool isCtfeValueValid(Expression newval)
     }
     if (newval.op == TOKnull)
     {
-        return tb.ty == Tnull || tb.ty == Tpointer || tb.ty == Tarray || tb.ty == Taarray || tb.ty == Tclass;
+        return tb.ty == Tnull || tb.ty == Tpointer || tb.ty == Tarray || tb.ty == Taarray || tb.ty == Tclass || tb.ty == Tdelegate;
```
Comment (UplinkCoder) [inline src/ddmd/ctfeexpr.d]: You might want to include Tfunction here as well.

Comment (WalterBright) [inline src/ddmd/ctfeexpr.d]: Tfunction is not a pointer, so that wouldn't work.

Comment (UplinkCoder) [inline src/ddmd/ctfeexpr.d]: Ah right ... Tfunction is the actual function and not the function-ptr.

Comment (UplinkCoder) [inline src/ddmd/ctfeexpr.d]: maybe we should do the same with delegate ?

**Comment by mathias-lang-sociomantic** [comment]
Stable please

**Comment by UplinkCoder** [comment]
@MartinNowak can you re-target this one to stable ?

**Comment by wilzbach** [comment]
> @MartinNowak can you re-target this one to stable ?

Everyone with merge rights can do this - it's not that hard. Just do sth like `git rebase --onto upstream/stable`

**Comment by UplinkCoder** [comment]
@wilzbach don't you have merge rights ?

**Comment by MartinNowak** [comment]
@WalterBright, please learn to target stable when making bugfixes. Even if it's not much work, constantly having to rebase PRs is kind of tiresome.

**Pull request merged**